### PR TITLE
Add integration tests for OpenAI NSQL functionality

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

PR adds `integration_models` tests that will include all tests related to AI/LLM functionality and adds integration test for `NSQL` functionality based on openai model.

`make test-integration-models` to run

## 🔨 Related Issues

Part of https://github.com/spiceai/spiceai/issues/3009

## 📄 Documentation Requirements

No

## 🤔 Concerns

Tests verify that the actual result for an NSQL query is correct (i.e., the row count is calculated correct). Since it is LLM-based and not deterministic, it may be unstable. As tests are not currently enabled for PRs, we can assess stability and decide if improvements are necessary.

Tests don't include case with `sample_data_enabled: true` as it is currently failing/investigating.

### CODE REVIEW

File: crates/runtime/tests/models/mod.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+use reqwest::Client;
+use serde_json::json;
+use spicepod::component::{
+    dataset::{acceleration::Acceleration, Dataset},
+    params::Params,
+};
+
+mod openai;
+
+fn get_taxi_trips_dataset() -> Dataset {
```
Comment (Jeadie) [inline crates/runtime/tests/models/mod.rs:26]: Question about this and integration tests in general: Why don't we define models/views/datasets etc from YAML strings?

File: crates/runtime/tests/models/openai.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    .await?
+    .contains(":10}"));
+
+    tracing::info!("/v1/nsql: Ensure error when invalid dataset name is provided");
+    assert!(send_nsql_request(
+        base_url.as_str(),
+        "how many records in taxi_trips dataset?",
+        Some("nql"),
+        Some(false),
+        Some(vec!["dummy".to_string()]),
```
Comment (Jeadie) [inline crates/runtime/tests/models/openai.rs]: nit: change `"dummy" -> "dataset_not_in_spice"` to make it instantly understandable.

File: crates/runtime/tests/models/mod.rs (hunk 1 above)
Comment (sgrebnov) [reply crates/runtime/tests/models/mod.rs:26]: This is just a more convenient and easy to maintain way to define such components. YAML is covered by E2E tests

File: crates/runtime/tests/models/openai.rs (hunk 2 above)
Comment (sgrebnov) [reply crates/runtime/tests/models/openai.rs]: Updated - thank you

File: crates/runtime/tests/models/openai.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    tracing::info!("/v1/nsql: Ensure default request succeeds");
+    assert!(send_nsql_request(
+        base_url.as_str(),
+        "how many records in taxi_trips dataset?",
+        None,
+        Some(false),
+        None,
+    )
+    .await?
+    .contains(":10}"));
```
Comment (sgrebnov) [inline crates/runtime/tests/models/openai.rs:84]: While working on `search` functionality tests I've improved this to return json and verify that json returned contains single record with value 10. The improvement will be submitted as part of next PR.

**Review by phillipleblanc — APPROVED** [review]

File: crates/runtime/tests/models/mod.rs (hunk 1 above)
Comment (sgrebnov) [reply crates/runtime/tests/models/mod.rs:26]: Jack - I plan to merge this and address this via separate PR if needed.

**Pull request merged**